flocker/node/agents/test/test_blockdevice.py

```diff
@@ -654,6 +717,74 @@ def test_match_configuration_to_state_of_datasets(self):
         self.assertEqual(expected_changes, actual_changes)
 
 
+class BlockDeviceDeployerResizeCalculateChangesTests(
+        SynchronousTestCase, ScenarioMixin
+):
+    """
+    Tests for ``BlockDeviceDeployer.calculate_changes`` in the cases relating
+    to resizing a dataset.
+    """
+    def test_maximum_size_increased(self):
+        """
+        ``BlockDeviceDeployer.calculate_changes`` returns a
+        ``ResizeBlockDeviceDataset`` state change operation if the
+        ``maximum_size`` of the configured ``Dataset`` is larger than the size
```

Comment:
Doesn't make a difference right now whether it's larger or smaller..  However, this branch only really tries to prove that the larger case works (and, indeed, the smaller case will not work yet).  There will probably be a `test_maximum_size_decreased` in an upcoming branch which makes a slightly different assertion.  (since resizefs/resize volume has to swap order).  So I guess I'll leave this.